File: frontend/src/app/core/services/submission.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Submission } from '../models/submission.models';

@Injectable({
    providedIn: 'root'
})
export class SubmissionService {
    private apiUrl = environment.apiBaseUrl + '/submissions';

    constructor(private http: HttpClient) { }

    getAllSubmissions(): Observable<Submission[]> {
        return this.http.get<Submission[]>(`${this.apiUrl}/`);
    }

    createSubmission(dto: Partial<Submission>): Observable<Submission> {
        return this.http.post<Submission>(`${this.apiUrl}/`, dto);
    }

    updateSubmission(id: string, dto: Partial<Submission>): Observable<Submission> {
        return this.http.put<Submission>(`${this.apiUrl}/${id}/`, dto);
    }

    deleteSubmission(id: string): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}/`);
    }
}

File: backend/src/main/java/com/ajay/career/service/SubmissionService.java
package com.ajay.career.service;

import com.ajay.career.dto.SubmissionDTO;

import java.util.List;
import java.util.UUID;

public interface SubmissionService {
    List<SubmissionDTO> getAllSubmissions();

    SubmissionDTO createSubmission(SubmissionDTO dto);

    SubmissionDTO updateSubmission(UUID id, SubmissionDTO dto);

    void deleteSubmission(UUID id);
}
